client/my-sites/checklist/main.jsx

```diff
@@ -28,6 +28,39 @@ import { isJetpackSite, getSiteSlug } from 'state/sites/selectors';
 import { recordTracksEvent } from 'state/analytics/actions';
 
 class ChecklistMain extends PureComponent {
+	componentDidMount() {
+		this.maybeRedirectJetpack();
+	}
+
+	componentDidUpdate( prevProps ) {
+		this.maybeRedirectJetpack( prevProps );
+	}
+
+	/**
+	 * Redirect Jetpack checklists to /plans/my-plan/:siteSlug
+	 *
+	 * Guard for relevant props updated and correct conditions
+	 *
+	 * @param {Object}   prevProps           Optional. Previous props from componentDidUpdate.
+	 * @param {?boolean} prevProps.isAtomic  Previous isAtomic
+	 * @param {?boolean} prevProps.isJetpack Previous isJetpack
+	 * @param {?string}  prevProps.siteSlug  Previous siteSlug
+	 */
+	maybeRedirectJetpack( { isAtomic, isJetpack, siteSlug } = {} ) {
+		if (
+			this.props.siteSlug &&
+			false === this.props.isAtomic &&
```

Comment:
hmm, b/c it's `null` or `undefined` while uninitialized?

Reply:
Yep, `null`:

https://github.com/Automattic/wp-calypso/blob/216e215f309187723acb1cc526123c499e92aaba/client/state/selectors/is-site-automated-transfer.js#L15-L17